rules/msk_topic_config_test.go

```diff
@@ -0,0 +1,100 @@
+package rules
+
+import (
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/terraform-linters/tflint-plugin-sdk/helper"
+)
+
+func Test_MskTopicConfigRule(t *testing.T) {
+	rule := &MskTopicConfigRule{}
+
+	const fileName = "topics.tf"
+	for _, tc := range []struct {
+		name     string
+		input    string
+		fixed    string
+		expected helper.Issues
+	}{
+		{
+			name: "topic without name",
+			input: `
+resource "kafka_topic" "topic_without_name" {
+}`,
+			expected: []*helper.Issue{
+				{
+					Rule:    rule,
+					Message: "missing replication_factor: it must be equal to '3'",
```

Comment:
should this alert about the missing name, not the replication factor?

Reply:
That's the thing, as the name is handled in the msk_topic_name. 
The thing is that if there is no name I couldn't find an elegant way to propose a fix (inserting the text in the block), thus we only emit the issue.

Reply:
> The name is handled in the msk_topic_name rule.

Can we just skip checking it here then and let that catch it?

>  The thing is that if there is no name I couldn't find an elegant way to propose a fix (inserting the text in the block), thus we only emit the issue.

But the emitted issue feels confusing: the plugin couldn't find a topic name, but spits out an error that the replication factor is wrong/missing

Reply:
So the logic in this plugin is:
if the replication factor is missing:
    if the name is present emit issue with suggested fix to put the replication factor after the name
    if the name is not present emit only the issue without a fix

>But the emitted issue feels confusing: the plugin couldn't find a topic name, but spits out an error that the replication factor is wrong/missing

So we're emitting the issue only when the replication factor is missing. It is not related to the topic name. We're checking here that if no topic name, no fixes are proposed, but the issue is still emited.

Reply:
> So we're emitting the issue only when the replication factor is missing

This is emitted in the `if !hasName` block?

Reply:
Yes, it is this part: https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/blob/topic-configs/rules/msk_topic_config.go#L99-L108

Reply:
Updated the scenario names in here: https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/12/commits/f86c05e33649fc5d8210dc7887b18a8bde3b2c75

Reply:
Ahhh I see now, my reading skills failed and I missed the `hasReplFactor` check just above that :facepalm: